lib/pbench/test/unit/server/query_apis/commons.py

```diff
@@ -0,0 +1,214 @@
+        return index
+
+    def test_non_accessible_user_data(self, client, server_config, pbench_token):
+        """
+        Test behavior when Authorization header does not have access to other user's data
+        """
+        # The pbench_token fixture logs in as user "drb"
+        # Trying to access the data belong to the user "pp"
+        self.json["user"] = "pp"
+        response = client.post(
+            f"{server_config.rest_uri}{self.pbench_endpoint}",
+            headers={"Authorization": "Bearer " + pbench_token},
+            json=self.json,
+        )
+        assert response.status_code == HTTPStatus.FORBIDDEN
+
+    @pytest.mark.parametrize(
+        "user", ("drb", "pp"),
+    )
+    def test_accessing_user_data_with_invalid_token(
+        self, client, server_config, pbench_token, user
+    ):
+        """
+        Test behavior when expired Authorization header provided
+        """
+        # valid token logout
+        response = client.post(
+            f"{server_config.rest_uri}/logout",
+            headers=dict(Authorization="Bearer " + pbench_token),
+        )
+        assert response.status_code == HTTPStatus.OK
+        self.json["user"] = user
+        response = client.post(
+            f"{server_config.rest_uri}{self.pbench_endpoint}",
+            headers={"Authorization": "Bearer " + pbench_token},
+            json=self.json,
+        )
+        assert response.status_code == HTTPStatus.FORBIDDEN
+
+    def test_missing_json_object(self, client, server_config, pbench_token):
+        """
+        Test behavior when no JSON payload is given
+        """
+        response = client.post(
+            f"{server_config.rest_uri}{self.pbench_endpoint}",
+            headers={"Authorization": "Bearer " + pbench_token},
+        )
+        assert response.status_code == HTTPStatus.BAD_REQUEST
+        assert response.json.get("message") == "Invalid request payload"
+
+    def test_missing_keys(self, client, server_config, keys, user_ok, pbench_token):
+        """
+        Test behavior when JSON payload does not contain all required keys.
+
+        Note that Pbench will silently ignore any additional keys that are
+        specified but not required.
+       """
+        response = client.post(
+            f"{server_config.rest_uri}{self.pbench_endpoint}",
+            headers={"Authorization": "Bearer " + pbench_token},
+            json=keys,
+        )
+        assert response.status_code == HTTPStatus.BAD_REQUEST
+        missing = [k for k in self.required_keys if k not in keys]
+        assert (
+            response.json.get("message")
+            == f"Missing required parameters: {','.join(missing)}"
+        )
+
+    def test_bad_dates(self, client, server_config, user_ok, pbench_token):
+        """
+        Test behavior when a bad date string is given
+        """
+        if not hasattr(self, "json_bad_date"):
+            pytest.skip("skipping the bad date test")
```

Comment:
As I said in my reply to the above comment, the initialization doesn't have to be done by a function named `__init__()`, eg., if PyTest reserves that, but it _should_ be done by **some** function in the base class rather than in the subclasses.

Reply:
Addressed in #2413.